docs/next/guides/building-and-testing/plugins.md

```diff
@@ -17,228 +15,242 @@ tags:
+   * The `enablePlugin` method is triggered on the `beforeInit` hook.
+   * It should contain your initial plugin setup, along with the hooks connections.
+   * Note, that this method is run only if the statement in the `isEnabled` method returns true.
    */
   enablePlugin() {
-    this.yourProperty = 'Your Value';
+    // Read the plugins' configuration from the initialization object.
+    this.configuration = this.getUnifiedConfig();
 
-    // Add all your plugin hooks here. It's a good idea to make use of the arrow functions to keep the context consistent.
+    // Add all your plugin hooks here. It's a good idea to use arrow functions to keep the plugin as a context.
     this.addHook('afterChange', (changes, source) => this.onAfterChange(changes, source));
 
-    // The super method assigns the this.enabled property to true, which can be later used to check if plugin is already enabled.
+    // The super method assigns the this.enabled property to true.
+    // It is a necessary step to update the plugin's settings properly.
     super.enablePlugin();
   }
 
   /**
-   * The disablePlugin method is used to disable the plugin. Reset all of your classes properties to their default values here.
+   * The disablePlugin method is used to disable the plugin.
    */
   disablePlugin() {
-    this.yourProperty = '';
-    this.anotherProperty = [];
+    // Reset all of your plugin class properties to their default values here.
+    this.configuration = null;
 
-    // The super method takes care of clearing the hook connections and assigning the 'false' value to the 'this.enabled' property.
+    // The `BasePlugin.disablePlugin` method takes care of clearing the hook connections
+    // and assigning the 'false' value to the 'this.enabled' property.
     super.disablePlugin();
   }
 
   /**
-   * The updatePlugin method is called on the afterUpdateSettings hook (unless the updateSettings method turned the plugin off).
-   * It should contain all the stuff your plugin needs to do to work properly after the Handsontable instance settings were modified.
+   * The updatePlugin method is called on the `afterUpdateSettings` hook
+   * (unless the updateSettings method turned the plugin off).
+   *
+   * It should contain all the stuff your plugin needs to do to work correctly
+   * after updating Handsontable instance settings.
    */
   updatePlugin() {
+    // The `updatePlugin` method needs to contain all the code needed to properly re-enable the plugin.
+    // In most cases simply disabling and enabling the plugin should do the trick.
+    const { enabled, msg } = this.getUnifiedConfig();
 
-    // The updatePlugin method needs to contain all the code needed to properly re-enable the plugin. In most cases simply disabling and enabling the plugin should do the trick.
-    this.disablePlugin();
-    this.enablePlugin();
+    // You can control if settings' update should go through disable->enable routine.
```

Comment:
// You can decide if updating the settings triggers the the disable->enable routine or not.